### DISCUSSION TITLE
How to define new units?

### DISCUSSION THREAD

**Discussion by milgner** [unitsofmeasurement/indriya#416] [Q&A]
In a project I stumbled upon units like `dpi` (dots per inch) and `lpi` (lines per inch), which are frequently used in printing.

Now I wonder how to integrate this with this library? A `dot`, at least in this context, doesn't have any concrete physical concept attached to it but it still should be possible to combine with physical units like `inch`.

Do I have to implement something based on `AbstractSystemOfUnits` like `PrintingUnits`? What base classes can I derive from or do I need to implement all the interfaces?

**Comment by keilw** [comment]
Please check other system libraries like 
- https://github.com/unitsofmeasurement/si-units
- https://github.com/unitsofmeasurement/uom-systems
- https://github.com/unitsofmeasurement/uom-domain (Domain-specific units including those for [Imaging/Print](https://github.com/unitsofmeasurement/uom-domain/tree/main/imaging) like `PIXEL` or `PIXEL_PER_INCH` (PPI) @milgner if you would like to contribute others like LPI or DPI, and `DOT`, we'd be more than happy about your contribution. Or use it as an inspiration for your own libraries.
